chore(l10n): improve checking of reviewer status to reduce redundant …

bcolsson opened the pull request.

xlisachan committed:
chore(deps): Update stripe to 10.7.0

chenba committed:
chore(signup): add FTC COPPA link

Because:
 - feedback from legal

This commit:
 - link to an explanation on FTC's web site for COPPA
 - update the input's label to refer to the new link

vbudhram committed:
feat(scopes): Allow trusted reliers to increase scope of token when using refresh token grant

StaberindeZA committed:
fix(coupons): pass promotion code on second sub
Because:

- When a customer, using PayPal, has an existing subscription, the
  promotion code is not included in the request to the backend.

This commit:

- Rework the onPaypalFormSubmit to match onStripeFormSubmit, so that
  all arguments are passed to the api call.

Closes #FXA-5878

flodolo commented on .github/workflows/l10n-gettext-extract.yml:
```diff
@@ -62,9 +62,29 @@ jobs:
-            # Add mozilla/fxa-l10n as reviewer for pull request if .pot file changes
-            echo "New changes found, adding reviewer."
-            gh pr edit $PR_NUMBER --add-reviewer mozilla/fxa-l10n
+             # get list of requested reviewers
```
Nit: at this point, this is a list of "pending" requested reviewers, since a reviewer who already approved doesn't show up here.

flodolo commented on .github/workflows/l10n-gettext-extract.yml:
```diff
@@ -62,9 +62,29 @@ jobs:
+            L10N_REVIEWERS=$(gh api -H "Accept: application/vnd.github+json" /orgs/mozilla/teams/fxa-l10n/members | jq -r '(.[].login | select(. != "clouserw"))')
+
+            # get list of usernames for existing reviews of PR
+            PR_REVIEWERS=$(gh pr view $PR_NUMBER --json reviews | jq -r ".reviews[].author.login")
```
This includes people who commented (like me on 14216), or even rejected.

bcolsson replied:
So the way I was thinking of this was to replicate how CODEOWNERS works, which only triggers once when the condition is met.
If we filter only by approved status that means a review request will get triggered when there's a push even after someone from l10n reviews and leaves a comment or requests changes. I'm assuming that we wouldn't be leaving comments or requesting changes to a PR unless we were already made aware of a ticket (most likely by a review request being triggered previously). 

That being said, thinking about it, in the above situations we'd then want to follow-up and review again so triggering the review again might be better for us than CODEOWNERS flow.

I'll go ahead and integrate your suggestions.

flodolo reviewed (commented):
Alternative approach, which looks simpler

```bash
# Check if fxa-l10n is still among the pending requested reviewers
gh pr view $PR_NUMBER --json reviewRequests | jq -r ".reviewRequests[].name" | grep -q "fxa-l10n"
if [ $? -ne 0 ]; then
    # Check if someone from the Localization Team already reviewed the PR.
    # Get the list of fxa-l10n reviewers, excluding non-l10n members
    L10N_REVIEWERS=$(gh api -H "Accept: application/vnd.github+json" /orgs/mozilla/teams/fxa-l10n/members | jq -r '(.[].login | select(. != "clouserw"))' | sort)
    # Get the list of reviewers who already approved the pull request
    PR_REVIEWERS=$(gh pr view $PR_NUMBER --json reviews | jq -r '.reviews[] | select(.state=="APPROVED") | .author.login' | sort)
    # Check intersection of the two lists
    L10N_APPROVERS=$(comm -12 <(echo "$L10N_REVIEWERS") <(echo "$PR_REVIEWERS"))
    if [ -z "$L10N_APPROVERS" ]; then
        # Add mozilla/fxa-l10n as reviewer
        echo "New changes found, adding reviewer."
        gh pr edit $PR_NUMBER --add-reviewer mozilla/fxa-l10n
    else
        echo "PR already reviewed by the localization team, skip adding reviewer."
    fi
else
    echo "fxa-l10n already added, skip adding reviewer."
fi
```

dependabot[bot] committed:
chore(deps): bump jose from 4.8.1 to 4.10.0

Bumps [jose](https://github.com/panva/jose) from 4.8.1 to 4.10.0.
- [Release notes](https://github.com/panva/jose/releases)
- [Changelog](https://github.com/panva/jose/blob/main/CHANGELOG.md)
- [Commits](https://github.com/panva/jose/compare/v4.8.1...v4.10.0)

---
updated-dependencies:
- dependency-name: jose
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

fxa-bananafox[bot] committed:
chore(deps): updated yarn.lock

dependabot[bot] committed:
chore(deps): bump apollo-server-express from 2.25.2 to 2.26.0

Bumps [apollo-server-express](https://github.com/apollographql/apollo-server/tree/HEAD/packages/apollo-server-express) from 2.25.2 to 2.26.0.
- [Release notes](https://github.com/apollographql/apollo-server/releases)
- [Changelog](https://github.com/apollographql/apollo-server/blob/apollo-server-express@2.26.0/CHANGELOG.md)
- [Commits](https://github.com/apollographql/apollo-server/commits/apollo-server-express@2.26.0/packages/apollo-server-express)

---
updated-dependencies:
- dependency-name: apollo-server-express
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

fxa-bananafox[bot] committed:
chore(deps): updated yarn.lock

chenba committed:
Merge pull request #14224 from mozilla/dependabot/npm_and_yarn/apollo-server-express-2.26.0

chore(deps): bump apollo-server-express from 2.25.2 to 2.26.0

vbudhram committed:
Merge pull request #14223 from mozilla/dependabot/npm_and_yarn/jose-4.10.0

chore(deps): bump jose from 4.8.1 to 4.10.0

vbudhram committed:
Merge pull request #14219 from mozilla/fxa-5880

feat(scopes): Allow trusted reliers to increase scope of token when using refresh token

chenba committed:
Merge pull request #14216 from mozilla/FXA-6011-why-coppa

chore(signup): add FTC COPPA link

bcolsson committed:
chore(l10n): improve checking of reviewer status to reduce redundant review requests

biancadanforth committed:
chore(payments): remove plan info and update PaymentConsentCheckbox copy

Because:

* Displaying the correct, localized amount and interval in more than one place is a source of discrepancies.
* This information is already shown in the PlanDetails component.

This commit:

* Removes localized amount and interval from the PaymentConsentCheckbox copy and updates the copy to reference the information in the PlanDetails component.

Closes #FXA-6049

StaberindeZA committed:
Merge pull request #14221 from mozilla/fxa-5878-missing-promo-from-paypal

fix(coupons): pass promotion code on second sub

biancadanforth committed:
Merge pull request #14229 from mozilla/FXA-6049

chore(payments): remove plan info and update PaymentConsentCheckbox copy

xlisachan committed:
Merge pull request #14065 from mozilla/update_stripe

chore(deps): Update stripe to 10.7.0

dschom committed:
task(content): Add support for tracing on client side

Because:
- We want to start trace capture at client side
- We can't report trace data directly to cloud trace

This Commit:
- Adds support for client side tracing in tracing/browser-tracing.ts
- Some refactors to fxa-shared/tracing to ensure code is reused
- Cleans up tests
- Introduces traceparent header and tracestate html headers
- Opens up OTLP ports on jaeger
- Introduces a clientName to trace config
- Adds an open telemetry collector service (otel-collector)
- Uses this services for all trace capture
- Adds flow id to trace id headers so traces can be validated

dschom committed:
Merge pull request #14222 from mozilla/FXA-6021/2-add-otel-collector

task(shared) - Add service to capture client side traces
